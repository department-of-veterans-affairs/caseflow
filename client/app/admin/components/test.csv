VET_LAST_NAME,VET_FIRST_NAME,VET_MIDDLE_NAME,VET_DATE_OF_BIRTH,VET_SSN,VET_PARTICIPANT_ID,VET_FILE_NUMBER,APPELLANT_LAST_NAME,APPELLANT_FIRST_NAME,APPELLANT_MIDDLE_NAME,APPELLANT_DATE_OF_BIRTH,APPELLANT_SSN,APPELLANT_GENDER,APPELLANT_ADDRESS,APPELLANT_PHONE,APPELLANT_EMAIL,APPELLANT_EDI_PI,APPELLANT_CORP_PID,APPELLANT_VACOLS_INTERNAL_ID,RELATIONSHIP_TO_VETERAN