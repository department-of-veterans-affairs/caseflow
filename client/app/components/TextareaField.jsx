import React from 'react';
import PropTypes from 'prop-types';
import classNamesFn from 'classnames';
import pluralize from 'pluralize';

export default class TextareaField extends React.Component {
  onChange = (event) => {
    this.props.onChange(event.target.value);
  }

  render() {
    const {
      errorMessage,
      hideLabel,
      id,
      maxlength,
      label,
      strongLabel,
      name,
      required,
      type,
      value,
      styling,
      textAreaStyling,
      disabled,
      labelStyling,
      placeholder,
      optional
    } = this.props;

    const className = 'cf-form-textarea' +
          `${errorMessage ? ' usa-input-error' : ''}`;

    // There is only a value for this variable if maxlength and value props are not null.
    // Otherwise characterLimitCount will be null also.
    const characterLimitCount = (Boolean(maxlength) && Boolean(value)) ? (maxlength - value.length) : null;

    const labelContents =
      <span>
        {label || name}
        {required && <span className="cf-required">Required</span>}
        {optional && <span className="cf-optional">Optional</span>}
      </span>;

    // hideLabel still leaves the label element in the DOM (for a11y purposes)
    // but makes it invisible to any screens
    return <div className={className} {...styling}>
      <label {...labelStyling}
        className={classNamesFn({ 'sr-only': hideLabel }, 'question-label')}
        htmlFor={id || name}>
        {
          strongLabel ?
            <strong>{labelContents}</strong> :
            labelContents
        }
      </label>
      {errorMessage && <span className="usa-input-error-message">{errorMessage}</span>}
      <textarea {...textAreaStyling}
        name={name}
        id={id || name}
        onChange={this.onChange}
        onKeyDown={this.props.onKeyDown}
        type={type}
        value={value}
        maxLength={maxlength}
        disabled={disabled}
        placeholder={placeholder}
      />
      { characterLimitCount !== null && <p>
        <i>{characterLimitCount} {pluralize('character', characterLimitCount)} left</i>
      </p> }
    </div>;
  }
}

TextareaField.defaultProps = {
  disabled: false,
  optional: false,
  required: false
};

TextareaField.propTypes = {
  hideLabel: PropTypes.bool,
  id: PropTypes.string,
  label: PropTypes.string,
  strongLabel: PropTypes.bool,
  maxlength: PropTypes.number,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onKeyDown: PropTypes.func,
  type: PropTypes.string,
  errorMessage: PropTypes.string,
  value: PropTypes.string,
  styling: PropTypes.object,
  disabled: PropTypes.bool,
  placeholder: PropTypes.string,
  optional: PropTypes.bool,
  required: PropTypes.bool,
  labelStyling: PropTypes.object,
  textAreaStyling: PropTypes.object
};
