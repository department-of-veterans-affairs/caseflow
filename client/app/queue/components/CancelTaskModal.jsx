import * as React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Link from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/Link';

import {
  taskById,
  appealWithDetailSelector
} from '../selectors';
import { onReceiveAmaTasks } from '../QueueActions';
import {
  requestPatch
} from '../uiReducer/uiActions';
import { taskActionData } from '../utils';
import TASK_STATUSES from '../../../constants/TASK_STATUSES';
import QueueFlowModal from './QueueFlowModal';

class CancelTaskModal extends React.Component {
  submit = () => {
    const {
      task,
      hearingDay
    } = this.props;
    const payload = {
      data: {
        task: {
          status: TASK_STATUSES.cancelled
        }
      }
    };

    const hearingScheduleLink = taskActionData(this.props).back_to_hearing_schedule ?
      <p>
        <Link href={`/hearings/schedule/assign?regional_office_key=${hearingDay.regionalOffice}`}>
          Back to Hearing Schedule
        </Link>
      </p> : null;
    const successMsg = {
      title: taskActionData(this.props).message_title,
      detail: <span><span>{taskActionData(this.props).message_detail}</span>{hearingScheduleLink}</span>
    };

    return this.props.requestPatch(`/tasks/${task.taskId}`, payload, successMsg).
      then((resp) => {
        this.props.onReceiveAmaTasks(resp.body.tasks.data);
      });
  }

  render = () => {
    const taskData = taskActionData(this.props);

    return <QueueFlowModal
      title={taskData ? taskData.modal_title : ''}
      pathAfterSubmit={(taskData && taskData.redirect_after) || '/queue'}
      submit={this.submit}
    >
      <div>{taskData && taskData.modal_body}</div>
    </QueueFlowModal>;
  };
}

CancelTaskModal.propTypes = {
  hearingDay: PropTypes.shape({
    regionalOffice: PropTypes.string
  }),
  onReceiveAmaTasks: PropTypes.func,
  requestPatch: PropTypes.func,
  task: PropTypes.shape({
    taskId: PropTypes.string
  })
};

const mapStateToProps = (state, ownProps) => ({
  task: taskById(state, { taskId: ownProps.taskId }),
  appeal: appealWithDetailSelector(state, ownProps),
  saveState: state.ui.saveState.savePending,
  hearingDay: state.ui.hearingDay
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  requestPatch,
  onReceiveAmaTasks
}, dispatch);

export default (withRouter(
  connect(mapStateToProps, mapDispatchToProps)(
    CancelTaskModal
  )
));
