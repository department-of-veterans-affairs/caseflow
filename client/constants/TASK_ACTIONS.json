{
  "ADD_ADMIN_ACTION": {
    "label": "Add admin action",
    "value": "colocated_task",
    "func": "add_admin_action_data"
  },
  "CHANGE_HEARING_DISPOSITION": {
    "label": "Change hearing disposition",
    "value": "modal/change_hearing_disposition"
  },
  "ASSIGN_OMO": {
    "label": "Assign OMO",
    "value": "omo_request/evaluate"
  },
  "ASSIGN_TO_ATTORNEY": {
    "label": "Assign to attorney",
    "value": "modal/assign_to_attorney",
    "func": "assign_to_attorney_data"
  },
  "ASSIGN_TO_HEARING_ADMIN_MEMBER": {
    "label": "Assign to person",
    "value": "modal/assign_to_hearing_admin_member",
    "func": "assign_to_user_data"
  },
  "ASSIGN_TO_MOTIONS_ATTORNEY": {
    "label": "Assign to person",
    "value": "modal/assign_to_person",
    "func": "assign_to_user_data"
  },
  "ASSIGN_TO_PERSON": {
    "label": "Assign to person",
    "value": "modal/assign_to_person",
    "func": "assign_to_user_data"
  },
  "ASSIGN_TO_PRIVACY_TEAM": {
    "label": "Assign to Privacy team",
    "value": "modal/assign_to_privacy_team",
    "func": "assign_to_privacy_team_data"
  },
  "ASSIGN_TO_TEAM": {
    "label": "Assign to team",
    "value": "modal/assign_to_team",
    "func": "assign_to_organization_data"
  },
  "SEND_MOTION_TO_VACATE_TO_JUDGE": {
    "label": "Send to judge",
    "value": "send_to_judge",
    "func": "send_motion_to_vacate_to_judge_data"
  },
  "ADDRESS_MOTION_TO_VACATE": {
    "label": "Address Motion to Vacate",
    "value": "address_motion_to_vacate",
    "func": "address_motion_to_vacate_data"
  },
  "REVIEW_VACATE_DECISION": {
    "label": "Decision ready for review",
    "value": "motion_to_vacate_checkout/review_vacatures"
  },
  "REVIEW_VACATE_RETURN_TO_JUDGE": {
    "label": "Return to judge",
    "value": "return_to_judge"
  },
  "SIGN_MOTION_TO_VACATE": {
    "label": "Sign Motion to Vacate",
    "value": "sign_motion_to_vacate",
    "func": "sign_motion_to_vacate_data"
  },
  "CANCEL_FOREIGN_VETERANS_CASE_TASK": {
    "label": "Cancel hearing request",
    "value": "modal/cancel_task",
    "func": "cancel_foreign_veterans_case_data"
  },
  "CANCEL_AND_RETURN_TASK": {
    "label": "Cancel task and return to judge",
    "value": "modal/cancel_task",
    "func": "cancel_task_data"
  },
  "CANCEL_TASK": {
    "label": "Cancel task",
    "value": "modal/cancel_task",
    "func": "cancel_task_data"
  },
  "CANCEL_ADDRESS_VERIFY_TASK_AND_ASSIGN_REGIONAL_OFFICE": {
    "label": "Cancel task and assign Regional Office",
    "value": "modal/cancel_task_and_assign_regional_office",
    "func": "cancel_address_verify_task_and_assign_regional_office_data"
  },
  "CHANGE_TASK_TYPE": {
    "label": "Change task type",
    "value": "modal/change_task_type",
    "func": "change_task_type_data"
  },
  "COLOCATED_RETURN_TO_ATTORNEY": {
    "label": "Send back to attorney",
    "value": "modal/mark_task_complete"
  },
  "COLOCATED_RETURN_TO_JUDGE": {
    "label": "Send back to judge",
    "value": "modal/mark_task_complete"
  },
  "COMPLETE_TRANSCRIPTION": {
    "label": "Mark as complete: Transcribed",
    "value": "modal/mark_task_complete",
    "func": "complete_transcription_data"
  },
  "CREATE_CHANGE_HEARING_DISPOSITION_TASK": {
    "label": "Send for hearing disposition change",
    "value": "modal/create_change_hearing_disposition_task"
  },
  "CREATE_CHANGE_PREVIOUS_HEARING_DISPOSITION_TASK": {
    "label": "Change previous hearing disposition",
    "value": "modal/create_change_hearing_disposition_task"
  },
  "CREATE_MAIL_TASK": {
    "label": "Create mail task",
    "value": "modal/create_mail_task",
    "func": "mail_assign_to_organization_data"
  },
  "JUDGE_LEGACY_CHECKOUT": {
    "label": "Ready for Dispatch",
    "value": "dispatch_decision/special_issues"
  },
  "JUDGE_AMA_CHECKOUT": {
    "label": "Ready for Dispatch",
    "value": "dispatch_decision/dispositions"
  },
  "JUDGE_CHECKOUT_PULAC_CERULLO_REMINDER": {
    "label": "Ready for Dispatch",
    "value": "modal/check_cavc"
  },
  "JUDGE_RETURN_TO_LIT_SUPPORT": {
    "value": "modal/return_to_lit_support",
    "func": "return_to_lit_support_data"
  },
  "JUDGE_RETURN_TO_ATTORNEY": {
    "label": "Return to attorney",
    "value": "modal/return_to_attorney",
    "func": "return_to_attorney_data"
  },
  "LIT_SUPPORT_PULAC_CERULLO": {
    "label": "Notify Pulac Cerullo team of CAVC Jurisdictional Conflict",
    "value": "modal/assign_to_pulac_cerullo",
    "func": "assign_to_pulac_cerullo_data"
  },
  "MARK_COMPLETE": {
    "label": "Mark task complete",
    "value": "modal/mark_task_complete",
    "func": "complete_data"
  },
  "MARK_NO_SHOW_HEARING_COMPLETE": {
    "label": "Mark task complete: Release case for review",
    "value": "modal/mark_task_complete",
    "func": "complete_data"
  },
  "PLACE_TIMED_HOLD": {
    "label": "Put task on hold",
    "value": "modal/place_timed_hold"
  },
  "END_TIMED_HOLD": {
    "label": "End hold early",
    "value": "modal/end_hold"
  },
  "SPECIAL_CASE_MOVEMENT": {
    "label": "Special Case Movement: Advance to judge",
    "value": "modal/special_case_movement",
    "func": "special_case_movement_data"
  },
  "TOGGLE_TIMED_HOLD": {
    "func": "toggle_timed_hold"
  },
  "REASSIGN_TO_JUDGE": {
    "label": "Re-assign to a judge",
    "value": "modal/reassign_to_judge",
    "func": "reassign_to_judge_data"
  },
  "REASSIGN_TO_PERSON": {
    "label": "Re-assign to person",
    "value": "modal/reassign_to_person",
    "func": "assign_to_user_data"
  },
  "RESCHEDULE_HEARING": {
    "label": "Mark as complete: Reschedule hearing",
    "value": "modal/cancel_task",
    "func": "assign_to_hearing_schedule_team_data"
  },
  "RESCHEDULE_NO_SHOW_HEARING": {
    "label": "Mark as complete: Postpone",
    "value": "modal/reschedule_hearing"
  },
  "QR_RETURN_TO_JUDGE": {
    "label": "Return to judge",
    "value": "modal/assign_to_judge",
    "func": "qr_return_to_judge_data"
  },
  "JUDGE_QR_RETURN_TO_ATTORNEY": {
    "label": "Return to attorney",
    "value": "modal/judge_qr_return_to_attorney",
    "func": "judge_qr_return_to_attorney_data"
  },
  "DISPATCH_RETURN_TO_JUDGE": {
    "label": "Return to judge",
    "value": "modal/dispatch_assign_to_judge",
    "func": "dispatch_return_to_judge_data"
  },
  "JUDGE_DISPATCH_RETURN_TO_ATTORNEY": {
    "label": "Return to attorney",
    "value": "modal/judge_dispatch_return_to_attorney",
    "func": "judge_dispatch_return_to_attorney_data"
  },
  "REVIEW_LEGACY_DECISION": {
    "label": "Decision ready for review",
    "value": "draft_decision/special_issues"
  },
  "REVIEW_AMA_DECISION": {
    "label": "Decision ready for review",
    "value": "draft_decision/dispositions"
  },
  "REVIEW_DECISION_DRAFT": {
    "func": "review_decision_draft"
  },
  "SCHEDULE_HEARING_SEND_TO_TEAM": {
    "label": "Hearing does need to be scheduled, create schedule hearing task",
    "value": "modal/send_colocated_task"
  },
  "SCHEDULE_VETERAN": {
    "label": "Schedule Veteran",
    "value": "modal/schedule_veteran",
    "func": "schedule_veteran_data"
  },
  "SEND_TO_SCHEDULE_VETERAN_LIST": {
    "label": "Send to Schedule Veterans list",
    "value": "modal/send_to_schedule_veterans_list",
    "func": "send_to_schedule_veterans_list"
  },
  "SEND_TO_TEAM": {
    "label": "Send to team",
    "value": "modal/send_colocated_task"
  },
  "SEND_TO_TRANSLATION": {
    "label": "Assign to Translation team",
    "value": "modal/assign_to_translation_team",
    "func": "assign_to_translation_team_data"
  },
  "SUBMIT_OMO_REQUEST_FOR_REVIEW": {
    "label": "Medical Request Ready for Review",
    "value": "omo_request/submit"
  },
  "WITHDRAW_HEARING": {
    "label": "Withdraw hearing",
    "value": "modal/cancel_task",
    "func": "withdraw_hearing_data"
  },
  "POSTPONE_HEARING": {
    "label": "Postpone hearing",
    "value": "modal/postpone_hearing",
    "func": "add_schedule_hearing_task_admin_actions_data"
  }
}
