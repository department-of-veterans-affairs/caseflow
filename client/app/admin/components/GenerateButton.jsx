/* eslint-disable func-style */
import React, { useEffect, useState } from 'react';
import Button from '../../components/Button';
import Modal from '../../components/Modal';
import Alert from '../../components/Alert';
import LoadingContainer from '../../components/LoadingContainer';
import { LOGO_COLORS } from 'app/constants/AppConstants';
import PropTypes from 'prop-types';
import { CSVLink } from 'react-csv';

const GenerateButton = (props) => {

  // state properties
  const [showConfirmModal, setShowConfirmModal] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [showEmptyResultsBanner, setShowEmptyResultsBanner] = useState(false);
  const [showErrorBanner, setShowErrorBanner] = useState(false);
  const [extractedResults, setExtractedResults] = useState('');

  useEffect(() => {
    setIsLoading(props.isLoading);
  }, [props.isLoading]);

  useEffect(() => {
    setExtractedResults(props.extractedResults);

    if (props.extractedResults?.length > 0) {
      setShowEmptyResultsBanner(false);
      setShowConfirmModal(true);
    }
  }, [props.extractedResults]);

  useEffect(() => {
    if (props.emptyResultsMessage?.length > 0) {
      setShowEmptyResultsBanner(true);
      setShowConfirmModal(false);
    }
  }, [props.emptyResultsMessage]);

  useEffect(() => {
    if (props.manualExtractionSuccess === false) {
      setShowErrorBanner(true);
    } else {
      setShowErrorBanner(false);
    }
  }, [props.manualExtractionSuccess]);

  const onClickGenerate = () => {
    props.sendExtractRequest();
  };

  return (
    <div style={{ height: '75vh' }}>
      {
        showEmptyResultsBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="No Veterans were found" type="success" />
        </div>
      }
      {
        showErrorBanner &&
        <div style={{ padding: '10px' }}>
          <Alert message="Veteran Extract Failed" type="error" />
        </div>
      }
      {
        !isLoading &&
        <Button
          id="generate-extract"
          onClick={() => onClickGenerate()}
        >
          Generate Veteran Extract
        </Button>
      }
      {
        showConfirmModal &&

        <Modal title="This file contains PII"
          confirmButton={<CSVLink data={extractedResults} filename="veteran_extract.csv" onClick={() => setShowConfirmModal(false)}>Confirm</CSVLink>}
          closeHandler={() => {
            setShowConfirmModal(false);
          }}
        >
          Are you sure you want to download it?
        </Modal>
      }
      {
        isLoading &&

        <LoadingContainer color={LOGO_COLORS.QUEUE.ACCENT}>
          <div className="loading-div">
            Action is Running...
          </div>
        </LoadingContainer>
      }
    </div>
  );
};

GenerateButton.propTypes = {
  extractedResults: PropTypes.string,
  manualExtractionSuccess: PropTypes.bool,
  isLoading: PropTypes.bool,
  sendExtractRequest: PropTypes.func,
  emptyResultsMessage: PropTypes.string,
};

export default GenerateButton;
