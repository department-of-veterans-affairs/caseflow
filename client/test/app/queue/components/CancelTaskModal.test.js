import React from 'react';
import { mount } from 'enzyme';

import { TextareaField } from 'app/components/TextareaField';
import CancelTaskModal from 'app/queue/components/CancelTaskModal';
import { queueWrapper } from '../../../data/stores/queueStore';
import {
  changeHearingRequestTypeTask,
  changeHearingRequestTypeTaskCancelAction
} from '../../../data/tasks';
import TASK_ACTIONS from 'constants/TASK_ACTIONS';

const requestPatchMock = jest.fn();

describe('CancelTaskModal', () => {
  test('for cancel change hearing request type action', () => {
    const cancelModal = mount(
      <CancelTaskModal
        requestPatch={requestPatchMock}
        taskId={changeHearingRequestTypeTask.uniqueId}
        match={{path: TASK_ACTIONS.CANCEL_CONVERT_HEARING_REQUEST_TYPE_TO_VIRTUAL.value}}
      />,
      {
        wrappingComponent: queueWrapper,
        wrappingComponentProps: {
          queue: {
            tasks: [
              {
                ...changeHearingRequestTypeTask,
                availableActions: [
                  changeHearingRequestTypeTaskCancelAction
                ]
              }
            ]
          },
          route: TASK_ACTIONS.CANCEL_CONVERT_HEARING_REQUEST_TYPE_TO_VIRTUAL.value
        }
      }
    );

    expect(cancelModal).toMatchSnapshot();
    expect(cancelModal.find(TextareaField)).toHaveLength(0);
  });
});
