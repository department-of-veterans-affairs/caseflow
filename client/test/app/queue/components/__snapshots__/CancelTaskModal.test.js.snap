// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelTaskModal for cancel change hearing request type action 1`] = `
<withRouter(Connect(CancelTaskModal))
  match={
    Object {
      "path": "modal/cancel_task",
    }
  }
  requestPatch={[MockFunction]}
  taskId="2"
>
  <Connect(CancelTaskModal)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "modal/cancel_task",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "modal/cancel_task",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "modal/cancel_task",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "modal/cancel_task",
        "url": "modal/cancel_task",
      }
    }
    requestPatch={[MockFunction]}
    taskId="2"
  >
    <CancelTaskModal
      hearingDay={
        Object {
          "hearingDate": null,
          "regionalOffice": null,
        }
      }
      highlightFormItems={false}
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "modal/cancel_task",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "modal/cancel_task",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "pathname": "modal/cancel_task",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "modal/cancel_task",
          "url": "modal/cancel_task",
        }
      }
      requestPatch={[Function]}
      task={
        Object {
          "addedByCssId": null,
          "appeal": Object {},
          "appealId": "123",
          "appealType": "LegacyAppeal",
          "assignedBy": Object {
            "cssId": null,
            "firstName": null,
            "lastName": null,
            "pgId": "1",
          },
          "assignedOn": "2020-09-08T10:02:49.210-04:00",
          "assignedTo": Object {
            "cssId": null,
            "id": "1",
            "isOrganization": true,
            "name": "Board of Veterans' Appeals",
            "type": "Bva",
          },
          "assigneeName": "Board of Veterans' Appeals",
          "availableActions": Array [
            Object {
              "data": Object {
                "message_detail": "You have successfully cancelled the convert hearing to virtual task",
                "message_title": "Task for Merlin V Langworth's case has been cancelled",
                "modal_title": "Cancel convert hearing to virtual task",
                "redirect_after": "/queue/appeals/1986897",
                "show_instructions": false,
              },
              "func": "cancel_convert_hearing_request_type_data",
              "label": "Cancel convert hearing to virtual",
              "value": "modal/cancel_task",
            },
          ],
          "availableHearingLocations": null,
          "cancelledBy": Object {
            "cssId": null,
          },
          "caseReviewId": null,
          "closedAt": "2020-09-08T14:13:33.878-04:00",
          "closestRegionalOffice": Object {
            "key": "RO17",
            "location_hash": Object {
              "alternate_locations": Array [
                "vba_317a",
                "vc_0742V",
                "vba_317",
                "vba_317",
                "vba_317",
              ],
              "city": "St. Petersburg",
              "facility_locator_id": "vba_317",
              "hold_hearings": true,
              "label": "St. Petersburg regional office",
              "state": "FL",
              "timezone": "America/New_York",
            },
          },
          "convertedBy": Object {
            "cssId": "BVASYELLOW",
          },
          "convertedOn": "2020-09-08T14:13:33.878-04:00",
          "createdAt": "2020-09-08T10:02:49.210-04:00",
          "decisionPreparedBy": null,
          "docketName": "hearing",
          "documentId": null,
          "externalAppealId": "23423534",
          "externalHearingId": "93897397",
          "hearingRequestType": null,
          "hideFromCaseTimeline": false,
          "hideFromQueueTableView": false,
          "hideFromTaskSnapshot": false,
          "instructions": Array [
            "Change hearing request type for a Travel Board hearing",
          ],
          "isFormerTravel": null,
          "isLegacy": false,
          "label": "Change hearing request type",
          "onHoldDuration": null,
          "parentId": "1",
          "placedOnHoldAt": null,
          "powerOfAttorneyName": null,
          "previousTaskAssignedOn": null,
          "startedAt": "2020-09-08T10:02:49.210-04:00",
          "status": "completed",
          "suggestedHearingLocation": null,
          "taskId": null,
          "timelineTitle": "Hearing type converted from Travel to Virtual",
          "type": "ChangeHearingRequestTypeTask",
          "uniqueId": "2",
          "workProduct": null,
        }
      }
      taskId="2"
    >
      <withRouter(Connect(QueueFlowModal))
        pathAfterSubmit="/queue/appeals/1986897"
        submit={[Function]}
        title="Cancel convert hearing to virtual task"
        validateForm={[Function]}
      >
        <Connect(QueueFlowModal)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "modal/cancel_task",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "modal/cancel_task",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "modal/cancel_task",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "modal/cancel_task",
              "url": "modal/cancel_task",
            }
          }
          pathAfterSubmit="/queue/appeals/1986897"
          submit={[Function]}
          title="Cancel convert hearing to virtual task"
          validateForm={[Function]}
        >
          <QueueFlowModal
            button="Submit"
            error={null}
            highlightInvalidFormItems={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "modal/cancel_task",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "modal/cancel_task",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "modal/cancel_task",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "modal/cancel_task",
                "url": "modal/cancel_task",
              }
            }
            pathAfterSubmit="/queue/appeals/1986897"
            resetSaveState={[Function]}
            saveSuccessful={null}
            submit={[Function]}
            submitDisabled={false}
            success={null}
            title="Cancel convert hearing to virtual task"
            validateForm={[Function]}
          >
            <Modal
              buttons={
                Array [
                  Object {
                    "classNames": Array [
                      "usa-button",
                      "cf-btn-link",
                    ],
                    "name": "Cancel",
                    "onClick": [Function],
                  },
                  Object {
                    "classNames": Array [
                      "usa-button-secondary",
                      "usa-button-hover",
                      "usa-button-warning",
                    ],
                    "disabled": false,
                    "loading": false,
                    "name": "Submit",
                    "onClick": [Function],
                  },
                ]
              }
              className=""
              closeHandler={[Function]}
              scrollLock={true}
              title="Cancel convert hearing to virtual task"
            >
              <section
                aria-describedby="modal_id-desc"
                aria-labelledby="modal_id-title"
                aria-modal="true"
                className="cf-modal active "
                id="modal_id"
                role="alertdialog"
              >
                <TouchProvider>
                  <ScrollLock
                    accountForScrollbars={true}
                  >
                    <Sheet
                      styles="body {
      box-sizing: border-box !important;
      overflow: hidden !important;
      position: relative !important;
      
      padding-right: 1024px !important;
    }"
                    />
                  </ScrollLock>
                </TouchProvider>
                <div
                  className="cf-modal-body"
                  id=""
                >
                  <button
                    className="cf-modal-close"
                    id="Cancel-convert-hearing-to-virtual-task-button-id-close"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="cf-icon-close"
                      height="55"
                      viewBox="0 0 55 55"
                      width="55"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <title>
                        close
                      </title>
                      <path
                        d="M52.6 46.9l-6 6c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-13-13-13 13c-.8.8-1.9 1.2-3 1.2s-2.2-.4-3-1.2l-6-6c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l13-13-13-13c-.8-.8-1.2-1.9-1.2-3s.4-2.2 1.2-3l6-6c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l13 13 13-13c.8-.8 1.9-1.2 3-1.2s2.2.4 3 1.2l6 6c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3l-13 13 13 13c.8.8 1.2 1.9 1.2 3s-.4 2.2-1.2 3z"
                      />
                    </svg>
                  </button>
                  <div
                    style={
                      Object {
                        "display": "flex",
                      }
                    }
                  >
                    <div
                      data-css-1gs0ko2=""
                    >
                      <h1
                        id="modal_id-title"
                      >
                        Cancel convert hearing to virtual task
                      </h1>
                      <div
                        data-css-o98ubo=""
                      />
                    </div>
                  </div>
                  <div
                    className="cf-modal-divider"
                  />
                  <div
                    className="cf-modal-controls"
                  >
                    <Button
                      classNames={
                        Array [
                          "usa-button",
                          "cf-btn-link",
                          "cf-push-left",
                        ]
                      }
                      id="Cancel-convert-hearing-to-virtual-task-button-id-0"
                      key="0"
                      linkStyling={false}
                      loading={false}
                      name="Cancel"
                      onClick={[Function]}
                      type="button"
                      willNeverBeLoading={false}
                    >
                      <span>
                        <button
                          className="usa-button cf-btn-link cf-push-left usa-button"
                          id="Cancel-convert-hearing-to-virtual-task-button-id-0"
                          onClick={[Function]}
                          type="button"
                        >
                          Cancel
                        </button>
                      </span>
                    </Button>
                    <Button
                      classNames={
                        Array [
                          "usa-button-secondary",
                          "usa-button-hover",
                          "usa-button-warning",
                          "cf-push-right",
                        ]
                      }
                      disabled={false}
                      id="Cancel-convert-hearing-to-virtual-task-button-id-1"
                      key="1"
                      linkStyling={false}
                      loading={false}
                      name="Submit"
                      onClick={[Function]}
                      type="button"
                      willNeverBeLoading={false}
                    >
                      <span>
                        <button
                          className="usa-button-secondary usa-button-hover usa-button-warning cf-push-right usa-button"
                          disabled={false}
                          id="Cancel-convert-hearing-to-virtual-task-button-id-1"
                          onClick={[Function]}
                          type="button"
                        >
                          Submit
                        </button>
                      </span>
                    </Button>
                  </div>
                </div>
              </section>
            </Modal>
          </QueueFlowModal>
        </Connect(QueueFlowModal)>
      </withRouter(Connect(QueueFlowModal))>
    </CancelTaskModal>
  </Connect(CancelTaskModal)>
</withRouter(Connect(CancelTaskModal))>
`;
